### PR TITLE
win_say: Clean up and check-mode support

### PR DESCRIPTION
**Pull request by dagwieers**
##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

win_say

##### ANSIBLE VERSION

v2.3

##### SUMMARY

Changes include:
- Clean up parameter handling
- Replace $result PSObject with a hash
- Added check-mode support

### CODE REVIEW

File: lib/ansible/modules/windows/win_say.ps1 [hunk 1]
```diff
@@ -19,59 +19,57 @@
...
+      Fail-Json $result "speech_speed needs to a integer in the range -10 to 10.  The value $speech_speed is outside this range."
    }
 }
 
-
 if ($msg_file -ne $null -and $msg -ne $null ) {
    Fail-Json $result "Please specify either msg_file or msg parameters, not both"
 }
 
-if ($msg_file -eq $null -and $msg -eq $null -and $start_sound_path -eq $null -and $end_sound_path -eq $null) {
```
Comment (jhawkesworth) [inline lib/ansible/modules/windows/win_say.ps1]: This check seems to have been removed - at least one of the 4 parameters 'msg', 'msg_file', 'start_sound_path' and 'end_sound_path' needs to be specified - I regard it as an error if none is specified.

**Comment by jhawkesworth** [comment]
Sorry, almost there!

See in line comment above

needs_revision

File: lib/ansible/modules/windows/win_say.ps1 (hunk 1 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_say.ps1]: RiIght, probably got squeeshed when I was investigating how to make mutual-exclusive and required options like we do in Python. What I was doing was needlessly complex so I dropped it. Still I think we need such functionality. I am sure @nitzmahone has some clear ideas about this too :-)

**Comment by dagwieers** [comment]
BTW The indentation was a combination of 2 spaces and 3 spaces. When I noticed some lines had 5 spaces I made it 6 spaces, but maybe you preferred 2 spaces over 3 spaces, I don't know.

**Comment by jhawkesworth** [comment]
Thanks for the fix.  

Yeah we definitely need some way of expressing mutually exclusive and one-of-many when it comes to handling module args.  There was a PR for a module which has some functions for this a while back but I guess it hasn't been merged (at least grepping for 'function' in lib/ansible/modules/windows/ hasn't found any likely looking candidates).

Regarding whitespace, I like it to be visually consistent within a file so what you've done works for me.

shipit

**Comment by dagwieers** [comment]
ready_for_review

**Comment by dagwieers** [comment]
bot_status

**Comment by jhawkesworth** [comment]
I don't understand  why 'maintainer_shipits' is 0 from the bot here.
Perhaps because I am in external committers group?

bot_broken

**Comment by dagwieers** [comment]
It also appears that in the review-process there are still some comments open ?
Not sure if this influences @ansibot as well.

**Pull request merged**